extensions/interactions/ItemSelectionInput/directives/ItemSelectionInputValidationService.js

```diff
@@ -148,6 +148,35 @@ oppia.factory('ItemSelectionInputValidationService', [
             return handledAnswer;
           });
         }
+        var answerChoiceToIndex = {};
+        seenChoices.forEach(function(seenChoice, choiceIndex) {
+          answerChoiceToIndex[seenChoice] = choiceIndex;
+        });
+
+        answerGroups.forEach(function(answerGroup, answerIndex) {
```

Comment:
This for-each loop is redundant. You can add `if (rule.type === 'IsProperSubsetOf')` condition in the above code only. In general, please try avoiding unnecessary duplicate code whenever possible. Sorry for not realising it earlier! Thanks!

Reply:
Hi @vibhor98 
Yeah, Ill remove it, I was going to remove it, but later saw that the for loop runs only one time, and instead of using ruleinputs[0], I thought Ill keep it the same way, any1 else trying to modify it later, wont need to go through it again. Ill remove it now. 
Thanks!

Reply:
The recent changes are wrong. It looks like you haven't understood the code properly yet. I meant moving this if condition entirely inside the for-each loop of line no: 121 (where all other rules are present) and there is no need of this redundant for-each loop.

Reply:
Hi! @vibhor98 , I am not sure, but based on what I understood,
The for loop, I removed by mistake thinking something else, Ill add it again
Also , 
 That for each loop (line 121 
```javascript
ruleInputs.forEach(function(ruleInput) { 
```
is inside an if loop,  
```javascript
 if (maxAllowedCount === 1) {
```
And this warning does not need maxAllowedCount to be 1.

Reply:
Oh, yes! I think I missed maxAllowedCount customisation arg. In that case, please add the loop that you've removed by mistake and everything else should be fine then.

Reply:
Done, added it.

Reply:
done,